ani-cli

```diff
@@ -1,131 +1,78 @@
 #!/bin/sh
 
-#---
-#dependencies:
-#  curl.se: ^7.8
-#  openssl.org: ^1.1
-#  ffmpeg.org: ^5.1
-#---
+version_number="4.0"
 
-# License preamble at the end of the file
-# Version number
-VERSION="3.4.7"
+# UI
 
-
-#######################
-# AUXILIARY FUNCTIONS #
-#######################
-
-help_text () {
-	while read -r line; do
-		printf "%s\n" "$line"
-	done <<-EOF
-
-	Usage:
-	  ${0##*/} [-f] [-s] [-v] [-x] [-q <quality>] [-a <episode>] [-d | -p <download_dir>] [<query>] [-r <provider_no.>]
-	  ${0##*/} [-f] [-s] [-v] [-x] [-q <quality>] -c
-	  ${0##*/} -h | -D | -U | -V
-
-	Options:
-	  -v use VLC as the media player
-	  -q set video quality (best|worst|360|480|720|1080)
-	  -s watch anime together with friends, using Syncplay (works with mpv only)
-	  -f use fzf for anime selection
-	  -a specify episode to watch
-	  -d download episode
-	  -p download episode to specified directory
-	  -c continue watching anime from history
-	  -h show helptext
-	  -D delete history
-	  -U fetch update from github
-	  -V print version number and exit
-	  -r select provider to scrape first [1-3]
-	  -x print all video links from all providers to stdout (for debugging purpose)
-
-	Episode selection:
-	  Multiple episodes can be chosen given a range
-	    Choose episode [1-13]: 1 6
-	    This would choose episodes 1 2 3 4 5 6
-		To select the last episode use -1
-
-	  When selecting non-interactively, the first result will be
-	  selected, if anime is passed
-EOF
+external_menu() {
+	rofi -sort -dmenu -i -width 1500 -p "$*"
 }
 
-retry () {
-	err "$*"
-	prompt
+launcher() {
+	[ "$use_external_menu" = "0" ] && fzf --prompt "$*"
+	[ "$use_external_menu" = "1" ] && external_menu "$*"
 }
 
-# display an error message to stderr (in red)
-err () {
-	printf "\33[2K\r\033[1;31m%s\033[0m\n" "$*" >&2
+nth() {
+	stdin=$(cat -)
+	[ -z "$stdin" ] && return 1
+	line_count="$(printf "%s\n" "$stdin" | wc -l)"
+	[ "$line_count" -eq 1 ] && printf "%s" "$stdin" && return 0
+	line=$(printf "%s" "$stdin" | sed 's|.*\t||' | launcher "$*")
+	[ -n "$line" ] && printf "%s" "$stdin" | grep -w "${line}$" || exit 1
```

Comment:
I assume this is where we construct our lines for usage in the fzf UI.
Perhaps we can integrate line numbers here via `nl`.

Reply:
seems reasonable